src/dashboard/apps/core/models.py

```diff
@@ -162,6 +169,44 @@ def get_validated_consents(self) -> QuerySet:
             "delivery_point__provider_assigned_id",
         )
 
+    def get_consents_stations_grouped(
+        self, consents: Optional[QuerySet["Consent"]] = None
+    ) -> list[Consent] | None:
+        """Get all consents for this entity with grouped stations."""
+        structured_consents: list = []
+
+        if not consents:
+            consents = self.get_consents()
+
+        for consent in consents:
+            stations_grouped = defaultdict(list)
+            for station in consent.delivery_point.stations.all().order_by(
```

Comment:
IMO a `DeliveryPoint` `def get_linked_stations(self) -> Dict` helper would be preferable to a `Consent.stations_grouped` attribute. That's odd for a model.